### PR TITLE
Feedback and command line button in nav flash on page load

### PR DESCRIPTION
**Pull request by gitstart-sourcegraph**
## Description
Fix Nav item flashing on page load (FeedBack button and WebCommandList Button)

https://user-images.githubusercontent.com/19534377/138307798-c247073e-670c-42c1-be5c-9cf73239c2cb.mov

## Refs

- Closes https://github.com/sourcegraph/sourcegraph/issues/26435
- [Gitstart Ticket](https://app.gitstart.com/clients/sourcegraph/tickets/SG-26435)
- https://www.loom.com/share/99ef01849bc44a3a86f8dc9c1e26e20b

### CODE REVIEW

**Review by valerybugakov — COMMENTED** [review]
@gitstart-sourcegraph, I see that the PR fixes the symptom of the issue, but what is the root cause of this flickering?

**Comment by gitstart-sourcegraph** [comment]
> @gitstart-sourcegraph, I see that the PR fixes the symptom of the issue, but what is the root cause of this flickering?

@valerybugakov For now, app needs to check `authenticatedUser` before showing `FeedbackPrompt` `NavAction`, so those elements are added into DOM a litter bit later than others + those elements have `transition` css rule (inherit from `.btn` => so we see the flickering effect

File: client/wildcard/src/components/NavBar/NavAction.module.scss [hunk 1]
```diff
@@ -39,6 +39,8 @@
     margin: 0;
     padding: 0;
     color: var(--icon-color);
+    //  Removes the delay in the visibility of the Element on page load
+    transition: visibility 0s;
```
Comment (valerybugakov) [inline client/wildcard/src/components/NavBar/NavAction.module.scss:43]: @gitstart-sourcegraph, the proposed fix breaks the transition effect on the element that should be preserved. Correct me if I'm wrong here.

**Review by valerybugakov — CHANGES_REQUESTED** [review]

File: client/wildcard/src/components/NavBar/NavAction.module.scss (hunk 1 above)
Comment (gitstart-sourcegraph) [reply client/wildcard/src/components/NavBar/NavAction.module.scss:43]: Thank @valerybugakov, my mistake, investigated deeply and found a reason for that here https://github.com/sourcegraph/sourcegraph/issues/26435#issuecomment-963422763

**Comment by gitstart-sourcegraph** [comment]
@valerybugakov Updated PR to fix the issue, basically we just to replace `useEffect` with `useMemo` to have these class toggle work before components get rendered

**Review by valerybugakov — APPROVED** [review]

**Pull request merged**